test/common/common/inline_map_test.cc

```diff
@@ -0,0 +1,224 @@
+
+    // The size will not change.
+    EXPECT_EQ(map->size(), 100);
+  }
+
+  // Lookup keys.
+  for (size_t i = 0; i < 100; ++i) {
+    EXPECT_EQ(*map->lookup("key_" + std::to_string(i)), "value_" + std::to_string(i) + "_new");
+
+    // The size will not change.
+    EXPECT_EQ(map->size(), 100);
+  }
+
+  // Lookup by typed inline handle.
+  for (size_t i = 0; i < handles.size(); ++i) {
+    const std::string key = "key_" + std::to_string(i);
+    auto handle = handles[i];
+    EXPECT_EQ(handle.inlineId(), i);
+    EXPECT_EQ(*map->lookup(handle), "value_" + std::to_string(i) + "_new");
+
+    // The size will not change.
+    EXPECT_EQ(map->size(), 100);
+  }
+
+  // Lookup non-existing keys.
+  EXPECT_FALSE(map->lookup("non_existing_key"));
+
+  // Remove keys by typed inline handle.
+  for (size_t i = 0; i < 10; ++i) {
+    const std::string key = "key_" + std::to_string(i);
+    auto handle = handles[i];
+    map->remove(handle);
+
+    EXPECT_EQ(map->size(), 100 - i - 1);
+  }
+
+  // Remove keys.
+  for (size_t i = 10; i < 100; ++i) {
+    map->remove("key_" + std::to_string(i));
+
+    EXPECT_EQ(map->size(), 100 - i - 1);
+  }
+
+  EXPECT_EQ(map->size(), 0);
+
+  // Lookup empty map by normal key will return nothing.
+  for (size_t i = 0; i < 100; ++i) {
+    EXPECT_EQ(map->lookup("key_" + std::to_string(i)), OptRef<std::string>{});
+  }
+
+  // Lookup empty map by typed inline handle will return nothing.
+  for (auto handle : handles) {
+    EXPECT_EQ(map->lookup(handle), OptRef<std::string>{});
+  }
+
+  // Operator[] will insert new entry if the key does not exist.
+  for (size_t i = 0; i < 10; ++i) {
+    EXPECT_EQ((*map)[handles[i]], "");
+
+    EXPECT_EQ(map->size(), i + 1);
+  }
+
+  for (size_t i = 10; i < 100; ++i) {
+    EXPECT_EQ((*map)["key_" + std::to_string(i)], "");
+
+    EXPECT_EQ(map->size(), i + 1);
+  }
+
+  EXPECT_EQ(map->size(), 100);
+
+  map.reset();
+}
+
+TEST(InlineMapWith20InlineKey, InlineMapWith20InlineKeyTestDestructWithEntries) {
+  InlineMapDescriptor<std::string> descriptor;
+
+  std::vector<InlineMapDescriptor<std::string>::Handle> handles;
+
+  // Create 20 inline keys.
+  for (size_t i = 0; i < 20; ++i) {
+    handles.push_back(descriptor.addInlineKey("key_" + std::to_string(i)));
+  }
+
+  // Add repeated inline keys will make no effect and return the same handle.
+  for (size_t i = 0; i < 20; ++i) {
+    EXPECT_EQ(handles[i].inlineId(),
+              descriptor.addInlineKey("key_" + std::to_string(i)).inlineId());
+  }
+
+  descriptor.finalize();
+
+  auto map = InlineMap<std::string, std::string>::create(descriptor);
+
+  // Insert inline entries.
+  for (size_t i = 0; i < 20; ++i) {
+    map->insert(handles[i], "value_" + std::to_string(i));
+  }
+
+  // Insert dynamic entries.
+  for (size_t i = 20; i < 100; ++i) {
+    map->insert("key_" + std::to_string(i), "value_" + std::to_string(i));
+  }
+
+  // Destruct the map.
+  map.reset();
+}
```

Comment:
can you add a new test with both unique_ptr<int> as key and as value, and make sure that you hit all the ways in which you can destruct an entry (erase, clear, overwrite, destruct map, operator= overwriting the map, anything else?)

Reply:
got it.

Reply:
Just remembered, the `absl:flat_hash_map` don't support to use unique_ptr as the key. See my old PR to absl https://github.com/abseil/abseil-cpp/pull/1292. (This PR isn't being merged, seems absl have no time to resolve this problem)

Reply:
nice, thanks.